### PR TITLE
[excel] (Comments) Creating a separate comments article

### PR DESCRIPTION
**Pull request by AlexJerabek**
The commenting feature has grown to the point a separate article is justified. This is that article.

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[3e340f5](https://github.com/OfficeDev/office-js-docs-pr/commits/3e340f5147171918e7b7215e03840461d2d670a3)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-comments.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-comments.md) | :warning:Warning | | [Details](#user-content-8d12538b-a67fb4f9)
[docs/reference/requirement-sets/excel-preview-apis.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/reference/requirement-sets/excel-preview-apis.md) | :warning:Warning | | [Details](#user-content-d94698b6-a67fb4f9)
[docs/excel/excel-add-ins-workbooks.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-workbooks.md) | :white_check_mark:Succeeded | |
[docs/images/excel-comments.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/images/excel-comments.png) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/toc.yml) | :white_check_mark:Succeeded | |

<a id="8d12538b-a67fb4f9"></a>
### [docs/excel/excel-add-ins-comments.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-comments.md)
 - **Line 170**: **[Warning]** `````Invalid file link:(~/excel/javascript/api/excel/excel.commentrichcontent).`````

<a id="d94698b6-a67fb4f9"></a>
### [docs/reference/requirement-sets/excel-preview-apis.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/reference/requirement-sets/excel-preview-apis.md)
 - **Line 20**: **[Warning]** `````Invalid link: '<a href="excel/excel-add-ins-workbooks.md#comments-preview">Comments</a>'. The file excel/excel-add-ins-workbooks.md doesn't contain a bookmark named 'comments-preview'.`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C10%5C16%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201910160014030259-1366%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=4c5UdGg8BwKwekH0N0z9pH4J7bgOP0i3TGqcl6L74Qk%3D&st=2019-10-16T00%3A12%3A39Z&se=2019-11-16T00%3A17%3A39Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[2c64c7c](https://github.com/OfficeDev/office-js-docs-pr/commits/2c64c7c4373d3146ae1ca9f54fb0d2a72ac66cca)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/requirement-sets/excel-preview-apis.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/reference/requirement-sets/excel-preview-apis.md) | :warning:Warning | | [Details](#user-content-d94698b6-af991f78)
[docs/excel/excel-add-ins-comments.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-comments.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-workbooks.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-workbooks.md) | :white_check_mark:Succeeded | |
[docs/images/excel-comments.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/images/excel-comments.png) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/toc.yml) | :white_check_mark:Succeeded | |

<a id="d94698b6-af991f78"></a>
### [docs/reference/requirement-sets/excel-preview-apis.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/reference/requirement-sets/excel-preview-apis.md)
 - **Line 20**: **[Warning]** `````Invalid file link:(~/excel/excel-add-ins-comments).`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C10%5C16%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201910161741457735-1366%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=2ZuT7x5LMECoP%2FBGu1GYAhUsQd7f3C5Zy66nVlrLuGo%3D&st=2019-10-16T17%3A40%3A18Z&se=2019-11-16T17%3A45%3A18Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[2036a73](https://github.com/OfficeDev/office-js-docs-pr/commits/2036a73ad6a52f375e9fd5e65acc240fae935f9a)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-comments.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-comments.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-workbooks.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-workbooks.md) | :white_check_mark:Succeeded | |
[docs/images/excel-comments.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/images/excel-comments.png) | :white_check_mark:Succeeded | |
[docs/reference/requirement-sets/excel-preview-apis.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/reference/requirement-sets/excel-preview-apis.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C10%5C16%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201910161744558506-1366%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=gI7zJ2llLqMeejBmlJ58Viwx2sg7HeRPCzXBmqmdHa0%3D&st=2019-10-16T17%3A44%3A02Z&se=2019-11-16T17%3A49%3A02Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-comments.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+---
+
+# Work with comments using the Excel JavaScript API (preview)
+
+> [!NOTE]
+> The comment APIs are currently available only in public preview. [!INCLUDE [Information about using preview APIs](../includes/using-excel-preview-apis.md)]
+
+This article describes how to add, read, modify, and remove comments in a workbook with the Excel JavaScript API. You can learn more about the comment feature from the [Insert comments and notes in Excel](https://support.office.com/article/insert-comments-and-notes-in-excel-bdcc9f5d-38e2-45b4-9a92-0b2b5c7bf6f8) article.
+
+In the Excel JavaScript API, a comment is both the initial note and the connected threaded discussion. It is tied to an individual cell. Every comment can be replied to by people viewing the workbook (that have sufficient permissions). A [Comment](/javascript/api/excel/excel.comment) object stores those replies as [CommentReply](/javascript/api/excel/excel.commentreply) objects. You should consider a comment to be a thread and that a thread must have a special entry as the starting point.
```
Comment (mscharlock) [inline docs/excel/excel-add-ins-comments.md]:
```suggestion
In the Excel JavaScript API, a comment is both the initial note and the connected threaded discussion. It is tied to an individual cell. Anyone viewing the workbook with sufficient permissions can reply to a comment. A [Comment](/javascript/api/excel/excel.comment) object stores those replies as [CommentReply](/javascript/api/excel/excel.commentreply) objects. You should consider a comment to be a thread and that a thread must have a special entry as the starting point.
```

Comment (mscharlock) [reply docs/excel/excel-add-ins-comments.md]: Not sure abou thte "special entry" phrasing - makes it seem like initial note has to be something super important but YMMV

File: docs/excel/excel-add-ins-comments.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    var comment = context.workbook.comments.getItemAt(0);
+    var reply = comment.replies.getItemAt(0);
+    reply.content = "Never mind";
+    return context.sync();
+});
+```
+
+## Delete comments
+
+To delete a comment use the `Comment.delete` method. Deleting a comment also deletes all the replies associated with that comment.
```
Comment (mscharlock) [inline docs/excel/excel-add-ins-comments.md]:
```suggestion
To delete a comment use the `Comment.delete` method. Deleting a comment also deletes the thread of replies associated with that comment.
```

File: docs/excel/excel-add-ins-comments.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+![An Excel comment, labelled "Comment" with two replies, labelled "Comment.replies[0]" and "Comment.replies[1].](../images/excel-comments.png)
+
+Comments within a workbook are tracked by the `Workbook.comments` property. This includes comments created by users and also comments created by your add-in. The `Workbook.comments` property is a [CommentCollection](/javascript/api/excel/excel.commentcollection) object that contains a collection of [Comment](/javascript/api/excel/excel.comment) objects. Comments are also accessible at the [Worksheet](/javascript/api/excel/excel.worksheet) level. The samples in this article work with comments at the workbook level, but they can be easily modified to use the `Worksheet.comments` property.
+
+## Add comments
+
+To add comments to a workbook, use the `CommentCollection.add` method, passing in the cell where the comment will be added, as either a string or [Range](/javascript/api/excel/excel.range) object, and the comment's text, as a string. Comments added by an add-in are attributed to the current user of that add-in.
```
Comment (mscharlock) [inline docs/excel/excel-add-ins-comments.md]: Optionally could make this bullets, e.g. 
Use the method, like so:
-Pass in the cell 
- Add comment text as string

File: docs/excel/excel-add-ins-comments.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    // Delete the first comment reply from this worksheet's first comment.
+    var comment = context.workbook.comments.getItemAt(0);
+    comment.replies.getItemAt(0).delete();
+    return context.sync();
+});
+```
+
+## Resolve comment threads
+
+A comment thread has a settable boolean value, `resolved`, to indicate if it is resolved. A value of `true` means the comment thread is resolved. A value of `false` means the comment thread is either new or reopened.
```
Comment (mscharlock) [inline docs/excel/excel-add-ins-comments.md]: Maybe configurable instead of settable?

**Comment by o365devx** [comment]
Docs Build status updates of commit _[4f969a3](https://github.com/OfficeDev/office-js-docs-pr/commits/4f969a37b07dbb34e196970357f912b91f04fae7)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-comments.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-comments.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-workbooks.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-workbooks.md) | :white_check_mark:Succeeded | |
[docs/images/excel-comments.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/images/excel-comments.png) | :white_check_mark:Succeeded | |
[docs/reference/requirement-sets/excel-preview-apis.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/reference/requirement-sets/excel-preview-apis.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C10%5C16%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201910161838173196-1366%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=j8C5Z%2FHexDYu7HOBu1eQTY9hnmXiR5sw%2B8FP%2BeJ41oI%3D&st=2019-10-16T18%3A37%3A07Z&se=2019-11-16T18%3A42%3A07Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by AlexJerabek** [comment]
Thanks for the feedback @mscharlock! I reworked the "Add comments" section a lot in response to your feedback (it was a little dated since mentions have been added).

**Comment by o365devx** [comment]
Docs Build status updates of commit _[36235e3](https://github.com/OfficeDev/office-js-docs-pr/commits/36235e3c243ee195b8b1d8c47c9dbbb7c736cedb)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-comments.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-comments.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-workbooks.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-workbooks.md) | :white_check_mark:Succeeded | |
[docs/images/excel-comments.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/images/excel-comments.png) | :white_check_mark:Succeeded | |
[docs/reference/requirement-sets/excel-preview-apis.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/reference/requirement-sets/excel-preview-apis.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C10%5C16%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201910161852495672-1366%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=vWU8m18x8AcTBeTF7hRatO2zMVy1VfiNpRiwENd8JMs%3D&st=2019-10-16T18%3A51%3A34Z&se=2019-11-16T18%3A56%3A34Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-comments.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+    comment.load(["authorEmail", "authorName", "creationDate"]);
+    return context.sync().then(function () {
+        console.log(`${comment.creationDate.toDateString()}: ${comment.authorName} (${comment.authorEmail})`);
+    });
+});
+```
+
+### Comment reply metadata
+
+Comment replies store the same types of metadata as the base comment.
```
Comment (mscharlock) [inline docs/excel/excel-add-ins-comments.md]:
```suggestion
Comment replies store the same types of metadata as the initial comment.
```

**Comment by o365devx** [comment]
Docs Build status updates of commit _[38fb168](https://github.com/OfficeDev/office-js-docs-pr/commits/38fb1680ef55c6c608aa9d7859da993eb35d344d)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-comments.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-comments.md) | :white_check_mark:Succeeded | |
[docs/excel/excel-add-ins-workbooks.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/excel/excel-add-ins-workbooks.md) | :white_check_mark:Succeeded | |
[docs/images/excel-comments.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/images/excel-comments.png) | :white_check_mark:Succeeded | |
[docs/reference/requirement-sets/excel-preview-apis.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/reference/requirement-sets/excel-preview-apis.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-Comments/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C10%5C16%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201910162206284692-1366%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=dfdlAonO9gu3SFUHuaTNAhgRLwtIiCJPoGO6SzHc2Fc%3D&st=2019-10-16T22%3A02%3A50Z&se=2019-11-16T22%3A07%3A50Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**